main.go

```diff
@@ -74,25 +76,89 @@ type customDNSProviderConfig struct {
 // solvers configured with the same Name() **so long as they do not co-exist
 // within a single webhook deployment**.
 // For example, `cloudflare` may be used as the name of a solver.
-func (c *customDNSProviderSolver) Name() string {
-	return "my-custom-solver"
+func (c *anexiaDNSProviderSolver) Name() string {
+	return "anexia"
+}
+
+// Find a TXT record with the given recordName and recordRData in the CloudDNS zone with the name zoneName
+func FindTXTRecord(c anxcloudClient.Client, ctx context.Context, zoneName string, recordName string, recordRData string) (*anxcloudZone.Record, error) {
+	zoneAPI := anxcloudZone.NewAPI(c)
+	records, err := zoneAPI.ListRecords(ctx, zoneName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range records {
+		// Work around ENGSUP-5257, CloudDNS API is inserting quotes into the TXT record data
+		if strings.HasPrefix(r.RData, "\"") && strings.HasSuffix(r.RData, "\"") {
```

Comment:
we wanted to include this in go-anxcloud instead, didn't we?

Reply:
Probably yes, but in the go-anxcloud version this depends on (which is the most recent one) that is not implemented. So for now we need this work around here, or it simply won't work.

In any case I tried to make this as specific as possible, in the hopes that it will be forward-compatible - if go-anxcloud starts stripping the double quotes, this should be a NOP for any sensible record data.

Reply:
Makes sense, ACK :)